_pages/implementation/implementation_API.md

```diff
@@ -0,0 +1,252 @@
+
+Clients SHALL add the following Spine proxy headers for audit and security purposes:
+
+- `Ssp-TraceID` - TraceID (generated per request) which identifies the sender's message/interaction (for example, a GUID/UUID).
+- `Ssp-From` - ASID which identifies the sender's FHIR endpoint.
+- `Ssp-To` - ASID which identifies the recipient's FHIR endpoint.
+- `Ssp-InteractionID` - identifies the FHIR interaction that is being performed <sup>1</sup>
+
+<sup>1</sup> please refer to the [Development - FHIR API guidance - operation guidance](integration_interaction_ids.html) for full details.
+
+The SSP SHALL perform the following checks to authenticate client request:
+
+- get the common name (CN) from the TLS session and compare the host name to the declared endpoint
+- check that the client/sending endpoint has been registered (and accredited) to initiate the given interaction
+- check that the server/receiving endpoint has been registered (and accredited) to receive/process the given interaction   
+
+#### Caching headers ####
+
+Providers SHALL use the following HTTP header to ensure that no intermediaries cache responses: `Cache-Control: no-store`
+
+### Managing resource contention ###
+
+To facilitate the management of [resource contention](http://hl7.org/fhir/STU3/http.html#concurrency), servers SHALL always return an `ETag` header with each resource including the resource’s `versionId`:
+
+```http
+HTTP 200 OK
+Date: Sat, 09 Feb 2013 16:09:50 GMT
+Last-Modified: Sat, 02 Feb 2013 12:02:47 GMT
+ETag: W/"23"
+Content-type: application/json+fhir
+```
+
+`ETag` headers which denote resource `version Id`s SHALL be prefixed with `W/` and enclosed in quotes, for example:
+
+```http
+ETag: W/"3141"
+```
+
+Clients SHALL submit update requests with an `If-Match` header that quotes the `ETag` from the server.
+
+```http
+PUT /Patient/347 HTTP/1.1
+If-Match: W/"23"
+```
+
+If the `version Id` given in the `If-Match` header does not match, the server returns a `409` **Conflict** status code instead of updating the resource.
+
+For servers that don't persist historical versions of a resource (that is, any resource other than the currently available/latest version) then they SHALL operate in line with the guidance provided in the following [Hay on FHIR - FHIR versioning with a non-version capable back-end](https://fhirblog.com/2013/11/21/fhir-versioning-with-a-non-version-capable-back-end/) blog post. This is to ensure that Care Connect servers will be compatible with version-aware clients, even though the server itself doesn't support the retrieval of historical versions.
+
+### Managing return errors ###
+
+To [manage return errors](http://hl7.org/fhir/STU3/http.html#2.1.0.4), FHIR defines an [OperationOutcome](http://hl7.org/fhir/STU3/operationoutcome.html) resource that can be used to convey specific detailed processable error information. See er_errorhandling.html for more information on errors
```

Comment:
Does this now need the errorhandling.html link?

Reply:
Fixed